indico/modules/events/papers/__init__.py

```diff
@@ -106,9 +106,16 @@ def _reviewing_area_visible(event):
             return False
         return event.cfp.can_access_reviewing_area(session.user)
 
+    def _call_for_papers_visible(event):
+        from indico.modules.events.contributions.util import has_contributions_with_user_as_submitter
+        if not session.user or not event.has_feature('papers'):
+            return False
+        return (has_contributions_with_user_as_submitter(event, session.user) or event.cfp.is_reviewer(session.user) or
+                event.cfp.is_judge(session.user))
```

Comment:
I'd add this method to the CFP class and use it instead of the last two criteria here:

```python
def is_staff(self, user):
    return self.is_manager(user) or self.is_judge(user) or self.is_reviewer(user)
```